dcc-portal-ui/app/scripts/survivalanalysis/js/survival-analysis.js

```diff
@@ -0,0 +1,302 @@
+      if (_.includes(disabledDataSets, data)) {
+        return;
+      }
+      var line = d3.svg.area()
+        .interpolate('step-after')
+        .x(function(p) { return x(p.x); })
+        .y(function(p) { return y(p.y); });
+
+      var setGroup = svg.append('g')
+        .attr('class', 'serie')
+        .attr('set-id', data.meta.id)
+        .attr('transform', 'translate(' + margin.left + ', ' + margin.top + ')');
+      var setColor = palette[i % palette.length];
+
+      // draw the data as an svg path
+      setGroup.append('path')
+        .datum(data.donors.map(function (d) { return {x: d.survivalTime, y: d.survivalEstimate}; }))
+        .attr('class', 'line')
+        .attr('d', line)
+        .attr('stroke', setColor);
+
+      // draw the data points as circles
+      setGroup.selectAll('circle')
+        .data(data.donors)
+        .enter()
+        .append('svg:circle')
+          .attr('class', 'point')
+          .attr('status', function (d) { return d.status; })
+          .attr('cx', function(d) { return x(d.survivalTime); })
+          .attr('cy', function(d) { return y(d.survivalEstimate); })
+          .attr('fill', setColor )
+          .on('mouseover', function (d) {
+            onMouseEnterDonor(d3.event, d);
+          })
+          .on('mouseout', function (d) {
+            onMouseLeaveDonor(d3.event, d);
+          })
+          .on('click', function (d) {
+            onClickDonor(d3.event, d);
+          })
+        ;
+    });
+
+    return svg;
+  }
+
+  var survivalAnalysisController = [
+    '$scope',
+    '$element',
+    function (
+      $scope,
+      $element
+    ) {
+      var ctrl = this;
+      var graphContainer = $element.find('.survival-graph').get(0);
+      var svg = d3.select(graphContainer).append('svg');
+      var tipTemplate = _.template($element.find('.survival-tip-template').html());
+
+      this.disabledDataSets = [];
+
+      var update = function () {
+        if (!ctrl.dataSets) {
+          return;
+        }
+        svg.selectAll('*').remove();
+        renderChart({
+          svg: svg, 
+          container: graphContainer, 
+          dataSets: ctrl.dataSets,
+          disabledDataSets: ctrl.disabledDataSets,
+          palette: ctrl.palette,
+          onMouseEnterDonor: function (event, donor) {
+            $scope.$emit('tooltip::show', {
+              element: event.target,
+              text: tipTemplate({
+                donor: _.extend(
+                  { isCensored: _.includes(ctrl.censoredStatuses,donor.status) },
+                  donor
+                ),
+                labels: ctrl.tipLabels,
+                unit: 'days'
+              }),
+              placement: 'right',
+              sticky: true
+            });
+          },
+          onMouseLeaveDonor: function () {
+            $scope.$emit('tooltip::hide');
+          },
+          onClickDonor: function (e, donor) {
+            window.open('/donors/'+donor.id, '_blank');
```

Comment:
Are you sure this is what we want to do? Usually we navigate to the page using the current window via location service (I think)